### PR TITLE
Added spheres union

### PR DESCRIPTION
**Pull request by matteomessmer**
### My change
Added the algorithm for calculating the volume of spheres union.

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by matteomessmer** [comment]
@Kush1101 do you have any update on my PR?

**Comment by CaedenPH** [comment]
> @Kush1101 do you have any update on my PR?

Kush seems to be an inactive user

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Personally, I preferred the f-strings, but everything else seems good

**Comment by matteomessmer** [comment]
@CaedenPH oh my mistake, another PR changed the strings and I overwrote that with the previous version

File: maths/volume.py [hunk 1]
```diff
@@ -107,6 +107,32 @@ def vol_spheres_intersect(
...
+    Calculate the volume of the union of two spheres that possibly intersect.
+    It is the sum of sphere A and sphere B minus their intersection.
+    First, it calculates the volumes (v1, v2) of the spheres,
+    then the volume of the intersection (i) and it returns the sum v1+v2-i.
+    If centers_distance is 0 then it returns the volume of the larger sphere
+    :return vol_sphere(radius_1) + vol_sphere(radius_2)
+                - vol_spheres_intersect(radius_1, radius_2, centers_distance)
+
+    >>> vol_spheres_union(2, 2, 1)
+    45.814892864851146
```
Comment (cclauss) [inline maths/volume.py:123]: A few more tests please including some with bad inputs. 0, negative numbers, floating point numbers, letters, None, etc.

File: maths/volume.py [hunk 2]
```diff
@@ -401,12 +427,13 @@ def main():
...
-    print(
-        f"Hollow Circular Cylinder: {vol_hollow_circular_cylinder(1, 2, 3) = }"
-    )  # ~= 28.3
     print(f"Conical Frustum: {vol_conical_frustum(2, 2, 4) = }")  # ~= 58.6
     print(f"Spherical cap: {vol_spherical_cap(1, 2) = }")  # ~= 5.24
     print(f"Spheres intersetion: {vol_spheres_intersect(2, 2, 1) = }")  # ~= 21.21
+    print(f"Spheres union: {vol_spheres_union(2, 2, 1) = }")  # ~= 45.81
+    print(
+        f"Hollow Circular Cylinder: {vol_hollow_circular_cylinder(1, 2, 3) = }"
+    )  # ~= 28.3
```
Comment (cclauss) [inline maths/volume.py:436]:
```suggestion
    #  ~= 28.3
    print(f"Hollow Circular Cylinder: {vol_hollow_circular_cylinder(1, 2, 3) = }")
```

**Comment by matteomessmer** [comment]
@cclauss I fixed some stuff and added more test

**Review by cclauss — APPROVED** [review]

**Pull request merged**